### PR TITLE
Add com.github.petebuffon.yafc

### PR DESCRIPTION
**Pull request by petebuffon**
### Please confirm your submission meets all the criteria

- [ ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ ] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ ] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.petebuffon.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  <url type="homepage">https://github.com/ShadowTheAge/yafc</url>
+  <screenshots>
+    <screenshot type="default">https://github.com/ShadowTheAge/yafc/blob/master/Docs/Media/Main.gif</screenshot>
+  </screenshots>
+  <launchable type="desktop-id">com.github.petebuffon.yafc.desktop</launchable>
+  <project_license>GPLv3</project_license>
```
Comment (hfiguiere) [inline com.github.petebuffon.appdata.xml:13]: This is an invalid license indentifier

https://spdx.dev/licenses/

File: com.github.petebuffon.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  <developer_name>ShadowTheAge</developer_name>
+  <releases>
+    <release version="0.5.8" date="2021-07-25"/>
+  </releases>
+</component>
```
Comment (hfiguiere) [inline com.github.petebuffon.appdata.xml:18]: content_rating is missing

File: com.github.petebuffon.yafc.desktop [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[Desktop Entry]
+Type=Application
+Name=YAFC
+Exec=YAFC
```
Comment (hfiguiere) [inline com.github.petebuffon.yafc.desktop:4]: It's missing "Icon", at least

File: com.github.petebuffon.yafc.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+app-id: com.github.petebuffon.yafc
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: YAFC
+finish-args:
+  - "--socket=wayland"
+  - "--device=dri"
+  - "--socket=x11"
```
Comment (hfiguiere) [inline com.github.petebuffon.yafc.yml:9]: does it really support wayland?

File: com.github.petebuffon.yafc.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  - "--share=ipc"
+  - "--filesystem=~/.factorio/mods:ro"
+  - "--filesystem=~/.steam/steam/steamapps/common/Factorio:ro"
+modules:
+  - name: yafc
+    buildsystem: simple
+    build-commands:
+      - install -Dp -m 755 YAFC /app/bin/YAFC
+      - install -Dp -m 644 com.github.petebuffon.yafc.desktop /app/share/applications/com.github.petebuffon.yafc.desktop
```
Comment (hfiguiere) [inline com.github.petebuffon.yafc.yml:18]: the appdata file is never installed.

File: com.github.petebuffon.yafc.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    sources:
+      - type: "file"
+        url: "https://github.com/petebuffon/yafc/releases/download/v0.5.8/YAFC"
```
Comment (hfiguiere) [inline com.github.petebuffon.yafc.yml:21]: so is this x86_64?

You need to https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.petebuffon.yafc.desktop (hunk 3 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.desktop:4]: There is no Icon, is that an issue?

File: com.github.petebuffon.yafc.yml (hunk 4 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.yml:9]: I've tested this program with wayland and it works.

Comment (hfiguiere) [reply com.github.petebuffon.yafc.yml:9]: then

```suggestion
  - "--socket=wayland"
  - "--device=dri"
  - "--socket=fallback-x11"
```

File: com.github.petebuffon.yafc.desktop (hunk 3 above)
Comment (hfiguiere) [reply com.github.petebuffon.yafc.desktop:4]: really?

Yes it is an issue. It is MANDATORY. No icon but a desktop file, it's a bug.

**Comment by hfiguiere** [comment]
BTW did you submit the .desktop and appdata file upstream? It's better when these are managed upstream (as well as icons)

Nothing specific to flathub / flatpak

**Comment by petebuffon** [comment]
The upstream developer focuses on Windows and doesn't currently have a .desktop file or appdata file, so I'm pretty much on my own for Linux packaging. I messaged them about the missing icon.

File: com.github.petebuffon.yafc.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.github.petebuffon.yafc</id>
```
Comment (barthalion) [inline com.github.petebuffon.yafc.appdata.xml:3]: Any reason you're using your username here?

File: com.github.petebuffon.yafc.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  <url type="homepage">https://github.com/ShadowTheAge/yafc</url>
+  <screenshots>
+    <screenshot type="default">https://github.com/ShadowTheAge/yafc/blob/master/Docs/Media/Main.gif</screenshot>
```
Comment (barthalion) [inline com.github.petebuffon.yafc.appdata.xml:10]: I don't think using GIF is going to work.

File: com.github.petebuffon.yafc.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  </screenshots>
+  <launchable type="desktop-id">com.github.petebuffon.yafc.desktop</launchable>
+  <project_license>GPL-3.0-only</project_license>
+  <developer_name>ShadowTheAge</developer_name>
+  <releases>
+    <release version="0.5.8" date="2021-07-25"/>
+  </releases>
+  <content_rating type="oars-1.0">
```
Comment (barthalion) [inline com.github.petebuffon.yafc.appdata.xml:18]: `none` can be skipped in `oars-1.1`.

File: com.github.petebuffon.yafc.yml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: YAFC
+finish-args:
+  - "--socket=wayland"
+  - "--device=dri"
+  - "--socket=fallback-x11"
+  - "--share=ipc"
+  - "--filesystem=~/.factorio/mods:ro"
+  - "--filesystem=~/.steam/steam/steamapps/common/Factorio:ro"
```
Comment (barthalion) [inline com.github.petebuffon.yafc.yml:12]: This misses flatpaked Steam.

**Comment by barthalion** [comment]
Please make sure to answer questions in your PR description.

File: com.github.petebuffon.yafc.yml (hunk 4 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.yml]: "--socket=fallback-x11" added

File: com.github.petebuffon.yafc.yml (hunk 5 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.yml:18]: appdata file is now installed

File: com.github.petebuffon.yafc.yml (hunk 6 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.yml:22]: Yex x86_64, flathub.json file added

File: com.github.petebuffon.yafc.appdata.xml (hunk 7 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.appdata.xml:3]: The upstream developer doesn't care about Linux support.

File: com.github.petebuffon.yafc.appdata.xml (hunk 8 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.appdata.xml]: Updated to PNG.

File: com.github.petebuffon.yafc.appdata.xml (hunk 9 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.appdata.xml]: Changed to <content_rating type="oars-1.1"/>

File: com.github.petebuffon.yafc.yml (hunk 10 above)
Comment (petebuffon) [reply com.github.petebuffon.yafc.yml]: Added - "--filesystem=xdg-run/app/com.valvesoftware.Steam:ro

**Comment by petebuffon** [comment]
Additionally I took out Wayland support after failed testing.

**Comment by barthalion** [comment]
bot, build com.github.petebuffon.yafc

**Comment by hfiguiere** [comment]
bot, build com.github.petebuffon.yafc

File: com.github.petebuffon.yafc.appdata.xml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  <url type="homepage">https://github.com/ShadowTheAge/yafc</url>
+  <developer_name>ShadowTheAge</developer_name>
+  <update_contact>pabuffon@gmail.com</update_contact>
+  <url type="bugtracker">https://github.com/petebuffon/yafc</url>
+  <launchable type="desktop-id">com.github.petebuffon.yafc.desktop</launchable>
+  <project_license>GPL-3.0-only</project_license>
+  <releases>
+    <release version="0.5.8" date="2021-07-25"/>
+  </releases>
```
Comment (hfiguiere) [inline com.github.petebuffon.yafc.appdata.xml:16]: missing `screenshots`

Comment (petebuffon) [reply com.github.petebuffon.yafc.appdata.xml:16]: Screenshots were added back.

File: com.github.petebuffon.yafc.appdata.xml [hunk 12]
```diff
@@ -6,12 +6,11 @@
...
   <url type="homepage">https://github.com/ShadowTheAge/yafc</url>
-  <screenshots>
-    <screenshot type="default">https://raw.githubusercontent.com/flathub/com.github.petebuffon.yafc/master/screenshots/demo_page.png</screenshot>
-  </screenshots>
+  <developer_name>ShadowTheAge</developer_name>
```
Comment (barthalion) [inline com.github.petebuffon.yafc.appdata.xml:9]: I think you need to make up your mind about this. Either you're the developer or the original author is and app ID needs to be changed.

Comment (petebuffon) [reply com.github.petebuffon.yafc.appdata.xml]: developer_name was changed to petebuffon

File: com.github.petebuffon.yafc.yml [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: YAFC
+finish-args:
+  - "--device=dri"
+  - "--socket=x11"
+  - "--share=ipc"
+  - "--filesystem=~/.factorio/mods:ro"
+  - "--filesystem=~/.steam/steam/steamapps/common/Factorio/data:ro"
+  - "--filesystem=xdg-run/app/com.valvesoftware.Steam:ro"
```
Comment (gasinvein) [inline com.github.petebuffon.yafc.yml:12]: You probably intended to access Steam flatpak data, not runtime dir.
```suggestion
  - "--filesystem=~/.var/app/com.valvesoftware.Steam:ro"
```

Comment (petebuffon) [reply com.github.petebuffon.yafc.yml]: Great, commit added.

**Comment by petebuffon** [comment]
I was on vacation and the invitation link expired before I could accept. Can you resend the invitation please?

**Comment by barthalion** [comment]
@petebuffon I've just sent out a new one.

**Pull request closed without merging**